docs/guides/dataweavetransformation.md

```diff
@@ -0,0 +1,311 @@
+* Deploy the `EventDisplay` service.
+* Deploy the `DataWeaveTransformation` object.
+* Configure the [Triggers](https://knative.dev/docs/eventing/broker/triggers/)
+* Deploy a curl pod that will allow us to send events to the broker.
+
+## How to use a DataWeaveTransformation
+
+A `DataWeaveTransformation` object can be configured to either reply to the event sender or to send the 
+transformed data to a `Sink`, if one is provided. In this guide, we will deploy without a `Sink` and 
+configure the replies from the transformation to route to the `EventDisplay` service using a `Broker` and a `Trigger`.
+
+The DataWeaveTransformation can have a pre-defined parameters configured in the yaml but it also allows to send the parameters as part of the request. In this guide we will use both ways, we will configure the pre-defined parameters in the yaml but we will also use other parameters in the request, which is made possible by enabling the allowPerEventDwSpell parameter.
+
+## DataWeaveTransformation parameters
+
+- allowPerEventDwSpell: Allow to send the DataWeaveSpell as part of the request. (Optional)
+- dwSpell: DataWeave spell used to transform incoming CloudEvents. (Optional)
+- inputContentType: Content type for transformation ['application/json', 'application/xml']. (Optional)
+- outputContentType: Content type for transformation output. ['application/json', 'application/xml']. (Optional)
+
+Below is a sample DataWeave spell that will be used throughout the guide.
+```
+%dw 2.0
+output application/json
+---
+{
+    email: payload.order.buyer.email,
+    name: payload.order.buyer.name,
+}
+```
+
+It transforms the following XML:
+```xml
+<order>
+	<product>
+		<price>5</price>
+		<model>Company 2020</model>
+	</product>
+	<item_amount>3</item_amount>
+	<payment>
+		<payment-type>credit-card</payment-type>
+		<currency>USD</currency>
+		<installments>1</installments>
+	</payment>
+	<buyer>
+		<email>james@hotmail.com</email>
+		<name>James</name>
+		<address>Cocodrile Boulevard 61</address>
+		<city>Seattle</city>
+		<state>CA</state>
+		<postCode>98101</postCode>
+		<nationality>USA</nationality>
+	</buyer>
+	<shop>main branch</shop>
+	<salesperson>Liam Smith</salesperson>
+</order>
+```
+
+Into this new JSON document:
+```json
+{
+  "email": "james@hotmail.com",
+  "name": "James"
+}
+```
+
+Let's go step by step to see how we can deploy this transformation as part of a TriggerMesh Bridge.
+
+Below is a diagram of the Bridge we will construct.
+
+![](../assets/images/dataweavetransformation.png)
+
+## Deploy the Broker
+Deploy a Broker by writing the following YAML in a file:
+```yaml
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  name: demo
+```
+
+Create the Broker with the following command:
+```console
+kubectl apply -f <manifest.yaml>
+```
+
+## Deploying the `EventDisplay` Service
+Let's now deploy the end of our event flow. The `EventDisplay` is a simple application that can be used to display CloudEvents. It can 
```

Comment:
```suggestion
Let's now deploy the Sink of our event flow. The `EventDisplay` is a simple application that can be used to display CloudEvents. It can 
```